lib/pages/goal_page.dart

```diff
@@ -3,102 +3,529 @@ import 'package:linear/model/goal.dart';
 import 'package:linear/pages/common_widgets/navbar.dart';
 import 'package:linear/pages/common_widgets/goal_widget.dart';
 import 'package:linear/util/apis.dart' as api;
-
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:linear/model/user.dart';
+import 'package:linear/pages/profile_page/profile_page.dart';
+import 'package:linear/model/goal.dart';
+ 
+// ignore: must_be_immutable
 class GoalPage extends StatefulWidget {
-  const GoalPage({super.key});
-
-  @override
-  State<GoalPage> createState() => GoalPageState();
+ const GoalPage({super.key});
+ 
+ 
+ @override
+ State<GoalPage> createState() => GoalPageState();
 }
-
+ 
 class GoalPageState extends State<GoalPage> {
-  List<dynamic> _goals = [];
+ 
+    List<dynamic> _allGoals= [];
+    List<dynamic> _goals = [];
+    String sortingBy = "Newest";
+    String filteringBy= "Open";
+    bool isSorting = true;
 
-  bool _isLoading = true;
-  bool _isErrorFetchingGoals = false;
+    TextEditingController searchInput = TextEditingController();
 
-  @override
-  void initState() {
-    super.initState();
-    fetchGoals();
-  }
+  List<dynamic> _searchResults = [];
+  bool _hasPerformedSearch = false;
+   bool _isSearching = false;
+ 
+ bool _isLoading = true;
+ bool _isErrorFetchingGoals = false;
+ 
+ @override
+ void initState() {
+   super.initState();
+   fetchGoals();
+ }
+ 
+ fetchGoals() {
+   final Future<Map<String, dynamic>> successfulMessage = api.getGoalsForGoalPage(context);
+   successfulMessage.then((response) {
+     if (response['status'] == true) {
+       setState(() {
+         _allGoals = response['goals'];
+         _isLoading = true;
+         _isErrorFetchingGoals = false;
+       });
+       setState(() {
+         _goals = _allGoals.toList();
+       });
```

Comment:
no need for a separate `setState`... and  no need to set `_goals` to `_allGoals` (see comment below why)